### ISSUE TITLE
support nonlocal statements

### ISSUE THREAD

**Issue by carljm** [astral-sh/ty#220]

**Comment by gusutabopb** [comment]
I ran into this today when using a closure to increment an integer from the outer scope.

Here's an MCVE for illustration purposes:

```python
def foo(n: int) -> int:
    count = 0

    def incr():
        nonlocal count
        count += 1

    for _ in range(n):
        incr()
    return count


if __name__ == "__main__":
    print(foo(10))
```

ty (0.0.1a6) reports this false positive:
```
error[unresolved-reference]: Name `count` used when not defined
```


https://play.ty.dev/12eb79c8-4504-43d0-a62c-2e99d8bfc067

**Comment by carljm** [comment]
we do support `global` now; repurposing this issue for `nonlocal` support

**Comment by oconnor663** [comment]
Support added in https://github.com/astral-sh/ruff/pull/19112.
